src/test/java/io/dropwizard/sharding/dao/locktest/LockTest.java

```diff
@@ -254,4 +255,288 @@ public void testPersist_alreadyExistingSame() throws Exception {
+        Assert.assertEquals(p2Id,lookupDao.get(p2Id).get().getMyId());
+        Assert.assertEquals("Parent 2",lookupDao.get(p2Id).get().getName());
+
+        //test partial update
+        final String childModifiedValue2 = "Hello Modified Partial";
+        final String parentModifiedValue2 = "Changed Partial";
+        lookupDao.lockAndGetExecutor(p1.getMyId())
+                .updateAll(relationDao, criteria, 0, 5, entities -> {
+                    final List<SomeOtherObject> modifiedObjects = new ArrayList<>();
+                    for (SomeOtherObject entity : entities) {
+                        if (entity.getId() == c2.getId()) {
+                            entity.setValue(childModifiedValue2);
+                            modifiedObjects.add(entity);
+                        }
+                    }
+
+                    return modifiedObjects;
+                })
+                .mutate(parent -> parent.setName(parentModifiedValue2))
+                .execute();
+
+
+        Assert.assertEquals(childModifiedValue,relationDao.get(p1.getMyId(), c1.getId()).get().getValue());
+        Assert.assertEquals(childModifiedValue2,relationDao.get(p1.getMyId(), c2.getId()).get().getValue());
+        Assert.assertEquals(parentModifiedValue2,lookupDao.get(p1Id).get().getName());
+        Assert.assertEquals(p1Id,lookupDao.get(p1Id).get().getMyId());
+
+        Assert.assertEquals("Hello3",relationDao.get(p2.getMyId(), c3.getId()).get().getValue());
+        Assert.assertEquals(p2Id,lookupDao.get(p2Id).get().getMyId());
+        Assert.assertEquals("Parent 2",lookupDao.get(p2Id).get().getName());
+    }
+
+    @Test
+    public void testSelectAndUpdateOrSave() throws Exception {
+        final String p1Id = "1";
+        final SomeLookupObject p1 = SomeLookupObject.builder()
+                .myId(p1Id)
+                .name("Parent 1")
+                .build();
+        lookupDao.save(p1);
+
+        final SomeOtherObject c1 = relationDao.save(p1.getMyId(), SomeOtherObject.builder()
+                .my_id(p1.getMyId())
+                .value("Hello")
+                .build()).get();
+
+
+        //test existing object
+        final String childModifiedValue = "Hello Modified";
+        final String parentModifiedValue = "Changed";
+        final DetachedCriteria criteria1 = DetachedCriteria.forClass(SomeOtherObject.class)
+                .add(Restrictions.eq("my_id", p1.getMyId()));
+
+        lookupDao.lockAndGetExecutor(p1.getMyId())
+                .selectAndUpdateOrSave(relationDao, criteria1, child -> {
+                    child.setValue("abcd");
+                    return child;
+                })
+                .mutate(parent -> parent.setName("xyzv"))
+                .selectAndUpdateOrSave(relationDao, criteria1, child -> {
+                    child.setValue(childModifiedValue);
+                    return child;
+                })
+                .mutate(parent -> parent.setName(parentModifiedValue))
```

Comment:
Why do this twice? the value saved will be the one that is finally present at end of txn.